In [61]:
# Read in given CSV - phq_all_final.csv - Renamed as GAD7
import numpy as np
import pandas as pd

df = pd.read_csv('/content/GAD7.csv', sep=',')
df

date  ...  score
0      2019-08-26T13:32:43.019162  ...      6
1      2019-11-15T17:55:11.364877  ...      1
2      2019-10-07T18:23:29.611351  ...     10
3      2019-11-04T20:16:28.610965  ...      7
4      2019-12-02T20:56:55.966354  ...      9
...                           ...  ...    ...
53693  2020-07-31T13:59:13.201741  ...      1
53694  2020-07-31T15:08:56.079027  ...      1
53695  2020-07-31T15:52:50.353265  ...      7
53696  2020-07-31T17:36:57.057913  ...      3
53697  2020-07-31T17:58:53.067462  ...      4

[53698 rows x 5 columns]

In [43]:
# Check data types for each column.
df.dtypes

date                    object
patient_id               int64
type                    object
patient_date_created    object
score                    int64
dtype: object

In [44]:
# Check data for missing values.
df.isnull().sum()

date                    0
patient_id              0
type                    0
patient_date_created    0
score                   0
dtype: int64

In [45]:
# Find how many patients there are in the dataset.
len(df.patient_id.unique())

15502

In [62]:
# Convert "date" and "patient_date_created" columns into a datetime format that can be used for visualization.
df['date'] = pd.to_datetime(df['date'], format='%Y-%m-%d %H:%M:%S.%f')

df['patient_date_created'] = pd.to_datetime(df['patient_date_created'], format='%Y-%m-%d %H:%M:%S.%f')
df

date  patient_id  ...       patient_date_created score
0     2019-08-26 13:32:43.019162        8430  ... 2019-06-03 13:23:02.635680     6
1     2019-11-15 17:55:11.364877        8430  ... 2019-06-03 13:23:02.635680     1
2     2019-10-07 18:23:29.611351       14338  ... 2019-06-03 13:32:30.668460    10
3     2019-11-04 20:16:28.610965       14338  ... 2019-06-03 13:32:30.668460     7
4     2019-12-02 20:56:55.966354       14338  ... 2019-06-03 13:32:30.668460     9
...                          ...         ...  ...                        ...   ...
53693 2020-07-31 13:59:13.201741       10683  ... 2020-07-31 13:54:20.164253     1
53694 2020-07-31 15:08:56.079027       12344  ... 2020-07-31 15:06:51.043730     1
53695 2020-07-31 15:52:50.353265       17475  ... 2020-07-31 15:40:07.726529     7
53696 2020-07-31 17:36:57.057913        9069  ... 2020-07-31 17:33:27.096572     3
53697 2020-07-31 17:58:53.067462        4870  ... 2020-07-31 17:58:21.004418     4

[53698 rows x 5 columns]

In [80]:
# Create bins to categorize each GAD7 score to severity level.
bins = [-1, 5, 11, 15, np.inf]

# Name each corresponding bin.
names = ['Low to Minimal', 'Mild','Moderate','Severe']

# Using pd.cut create a new dataframe based on the bins created for severity level.
df['severity_level'] = pd.cut(df['score'], bins, labels=names)
df

date  patient_id  ... score  severity_level
0     2019-08-26 13:32:43.019162        8430  ...     6            Mild
1     2019-11-15 17:55:11.364877        8430  ...     1  Low to Minimal
2     2019-10-07 18:23:29.611351       14338  ...    10            Mild
3     2019-11-04 20:16:28.610965       14338  ...     7            Mild
4     2019-12-02 20:56:55.966354       14338  ...     9            Mild
...                          ...         ...  ...   ...             ...
53693 2020-07-31 13:59:13.201741       10683  ...     1  Low to Minimal
53694 2020-07-31 15:08:56.079027       12344  ...     1  Low to Minimal
53695 2020-07-31 15:52:50.353265       17475  ...     7            Mild
53696 2020-07-31 17:36:57.057913        9069  ...     3  Low to Minimal
53697 2020-07-31 17:58:53.067462        4870  ...     4  Low to Minimal

[53698 rows x 6 columns]

In [83]:
# Count number of severity level responses total in dataframe.
df[('severity_level')].value_counts()


Low to Minimal    35402
Mild              11429
Moderate           3497
Severe             3370
Name: severity_level, dtype: int64

**Given the information you have and any light research you’d like to do on the topic, what insights can you draw?**

From this dataset, there are 15,502 patients seeing mental health providers. There are more "Low to minimal" responses within the dataset with the number of "Mild", "Moderate", and "Severe" responses declining. There were likely more severe GAD7 scores to begin with, and as patients went to appointments with mental health professionals, the number of "Moderate", "Mild", and "Low to Minimal" severity levels accumulated as the patients felt better.

**What assumptions have you made about the data?**

Each time a patient visits their mental health provider, their score from the GAD7 will go down.

**What are 2-3 additional pieces of information that would be important to collect?**

1. The proposed treatment actions after each assessment.
2. The number of meetings the patient has had with the mental health provider.
3. Age and Gender of the patients.